File: xap-drools-integration-master/common/src/main/java/com/c123/demo/real/BaseWithdrawFact.java
package com.c123.demo.real;

import java.io.Serializable;
import java.util.UUID;

public class BaseWithdrawFact extends BaseAccountOperationFact implements Serializable{

	private Integer offering;
    private Integer fundsType;
    private Integer operationSourceApplication;
    private String requestCurrencyCode;
    private Integer updateBalanceReason;
    
    
	public BaseWithdrawFact() {
		super();
	}
    
	public Integer getOffering() {
		return offering;
	}
	public void setOffering(Integer offering) {
		this.offering = offering;
	}
	public Integer getFundsType() {
		return fundsType;
	}
	public void setFundsType(Integer fundsType) {
		this.fundsType = fundsType;
	}
	public Integer getOperationSourceApplication() {
		return operationSourceApplication;
	}
	public void setOperationSourceApplication(Integer operationSourceApplication) {
		this.operationSourceApplication = operationSourceApplication;
	}
	public String getRequestCurrencyCode() {
		return requestCurrencyCode;
	}
	public void setRequestCurrencyCode(String requestCurrencyCode) {
		this.requestCurrencyCode = requestCurrencyCode;
	}

	public Integer getUpdateBalanceReason() {
		return updateBalanceReason;
	}

	public void setUpdateBalanceReason(Integer updateBalanceReason) {
		this.updateBalanceReason = updateBalanceReason;
	}

	@Override
	public String toString() {
		return "BaseWithdrawFact [offering=" + offering + ", fundsType="
				+ fundsType + ", operationSourceApplication="
				+ operationSourceApplication + ", requestCurrencyCode="
				+ requestCurrencyCode + ", updateBalanceReason="
				+ updateBalanceReason + "]";
	}


   
}

File: Drools_client/solution/Drools.Common/Entities/BaseWithdrawFact.cs
using GigaSpaces.Core.Metadata;
using System;

namespace Drools.Common.Entities
{
    [Serializable]
    [SpaceClass(AliasName = "com.c123.demo.real.BaseWithdrawFact")]
    public class BaseWithdrawFact : BaseAccountOperationFact
    {
        public BaseWithdrawFact()
        {

        }

        public BaseWithdrawFact(BaseWithdrawFact originFact)
            : base(originFact)
        {
            FundsType = originFact.FundsType;
            Offering = originFact.Offering;
            OperationSourceApplication = originFact.OperationSourceApplication;
            RequestCurrencyCode = originFact.RequestCurrencyCode;
            UpdateBalanceReason = originFact.UpdateBalanceReason;
        }
        [SpaceProperty(AliasName = "offering")]
        public int? Offering { get; set; }
        [SpaceProperty(AliasName = "fundsType")]
        public FundsType? FundsType { get; set; }
        [SpaceProperty(AliasName = "updateBalanceReason")]
        public int? UpdateBalanceReason { get; set; }
        [SpaceProperty(AliasName = "operationSourceApplication")]
        public int? OperationSourceApplication { get; set; }
        [SpaceProperty(AliasName = "requestCurrencyCode")]
        public string RequestCurrencyCode { get; set; }

        // To verify that we will not duplicate facts, this is the unique definition of a fact.
        protected override object[] IdComponents
        {
            get
            {
                return new object[]
                {
                    this.NetworkID, this.RequestReference
                };
            }
        }
    }
}
